code/modules/antagonists/wizard/equipment/spellbook.dm

```diff
@@ -229,12 +229,24 @@
 	cost = 3
 	no_coexistance_typecache = /obj/effect/proc_holder/spell/targeted/infinite_guns/arcane_barrage
 
+/datum/spellbook_entry/infinite_guns/Refund(mob/living/carbon/human/user, obj/item/spellbook/book)
+	for (var/obj/item/currentItem in user.get_all_gear())
+		if (istype(currentItem, /obj/item/gun/ballistic/rifle/boltaction/enchanted))
```

Comment:
if you somehow get both an enchanted bolt action rifle and an arcane barrage hand on your person at the same time, refunding arcane barrage will only delete the arcane barrage hand, but refunding lesser summon guns will delete both the enchanted bolt action rifle and the arcane barrage hand.

Reply:
> how does this fix #47796?
> 
> also, this kind of thing is why I'm reluctant to talk about my gamer tricks before I use them at least once.
> 
> s'pose it was inevitable, though.

This solves the problem by removing the enchanted rifle or arcane barrage spell from your hand if equipped
It also makes it so that dropping the enchanted rifle removes its magazine / chambered bullet
Dropping arcane barrage already deletes it

Reply:
Fixed, forgot that istype also includes subtypes